src/main/drivers/compass/compass_hmc5883l.c

```diff
@@ -232,21 +296,16 @@ static bool hmc5883lInit(void)
     }
 
     // Apply the negative bias. (Same gain)
-#ifdef USE_MAG_SPI_HMC5883
-    hmc5883SpiWriteCommand(HMC58X3_R_CONFA, 0x010 + HMC_NEG_BIAS);   // Reg A DOR = 0x010 + MS1, MS0 set to negative bias.
-#else
-    i2cWrite(MAG_I2C_INSTANCE, MAG_ADDRESS, HMC58X3_R_CONFA, 0x010 + HMC_NEG_BIAS);   // Reg A DOR = 0x010 + MS1, MS0 set to negative bias.
-#endif
+
+    hmc5883lWriteRegister(HMC58X3_R_CONFA, 0x010 + HMC_NEG_BIAS);   // Reg A DOR = 0x010 + MS1, MS0 set to negative bias.
+
     for (i = 0; i < 10; i++) {
-#ifdef USE_MAG_SPI_HMC5883
-        hmc5883SpiWriteCommand(HMC58X3_R_MODE, 1);
-#else
-        i2cWrite(MAG_I2C_INSTANCE, MAG_ADDRESS, HMC58X3_R_MODE, 1);
-#endif
+        hmc5883lWriteRegister(HMC58X3_R_MODE, 1);
         delay(50);
```

Comment:
Measurement should be ready in 6ms (typ). 20 x 50ms delay is 1s, non-negligible initialization time ...

Reply:
Reduce to 20 for now. We may go down to 10 if someone can do testing.